component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java

```diff
@@ -579,43 +824,46 @@ public void publish(Object payload, DynamicOptions dynamicOptions) {
                     .getBytes(Charset.defaultCharset()))));
         }
         cMessage.completeMessage();
-        clientConnector.send(cMessage);
+        CountDownLatch latch = new CountDownLatch(1);
+        DefaultListener listener = new DefaultListener(latch, authType);
+        HttpResponseFuture responseFuture = clientConnector.send(cMessage);
+        responseFuture.setHttpConnectorListener(listener);
+
+        if (HttpConstants.OAUTH.equals(authType)) {
+            try {
+                boolean latchCount = latch.await(30, TimeUnit.SECONDS);
+                if (!latchCount) {
+                    log.debug("Time out due to getting new access token. ");
```

Comment:
We are using this 'sendRequest` method for 
1. getting access token. 
2. sending event to the HTTP endpoint.

In case 2, why do we need to wait on the latch and print this log?